added grep command info

ahyerman opened the pull request:
Closes #13

mterwill commented on _commands/basics/grep.md:
```diff
@@ -0,0 +1,62 @@
+grep
+-------
+
+`grep` is used to search files for the occurence os string of characters matching a special pattern (sometimes called a regex or regular expression).
```
Typo? Also note that grep doesn't support full regex – that's egrep. Would make note somewhere in here.

mterwill commented:
I would also like to see an example of grep used to filter another (simple) command, as that could definitely benefit the target audience of this reference. Even something as simple as  ``ls -l | grep cpp``

ahyerman commented:
fixed spelling typo. Added ls -l example. Using grep as a filter for commands to search for files. Explained process could be used to find processes as well. Linked grep to ls page.

themoh commented:
A small section on the -e flag/egrep would be useful

ahyerman commented:
Added recommended changes with an example of using egrep. Also removed regex from the top example to instead say string.

themoh commented:
Looks really good so far. Add in a section on the -r flag also, and I think you should be good. @mterwill any other suggestions?

ahyerman commented:
Added a section with the -r flag along with an example.

ppannuto commented:
Looks good, thanks!
